### PR TITLE
Create Readme.md file

### PR DESCRIPTION
**Pull request by framos-gemini**
This version contains a new README.md file describing how to deploy the gmp and giapi-glue.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+# Desployment from binaries. 
+It is possible deploying the GMP and GIAPI-GLUEc++ from their binaries. Gemini recommend uses Rocky 8 or Centos 7 because these are the Operating Systems supported by GEMINI and for which the GIAPI-GLUEc++ binaries have been compiled. 
+
+If you want to deploy the system in other platform, you should compile the external libraries of the [GIAPI-GLUEc++](https://github.com/gemini-hlsw/giapi-glue-cc/tree/develop/external) for you platform. In the previous link is decribed the instrucction executed by GEMINI to CENTOS7 and ROCKY8.
+
+## Dependencies 
+The only dependencies we need for GMP and GIAPI-glue are as follows.
+* Java JDK 1.8
+* GCC greater than 4.8.1.
```
Comment (cquiroz) [inline README.md:18]: Should it be a newer gcc that supports C++11?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
+# GIAPI OVERVIEW
+The GIAPI concept was developed by GEMINI to facilitate the interaction between the Telescope and the instrument software. Thanks to GIAPI, instrument developers do not need to know in detail the GEMINI Telescope Control System. They only need to deploy the GMP (Gemini Master Process) and incorporate GIAPI-GLUE in their source code to be able to communicate with the Telescope.
```
Comment (cquiroz) [inline README.md:2]: I think GEMINI should not be capitalize
`s/GEMIINI/Gemini/g`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
+# GIAPI OVERVIEW
+The GIAPI concept was developed by GEMINI to facilitate the interaction between the Telescope and the instrument software. Thanks to GIAPI, instrument developers do not need to know in detail the GEMINI Telescope Control System. They only need to deploy the GMP (Gemini Master Process) and incorporate GIAPI-GLUE in their source code to be able to communicate with the Telescope. 
+
+The GMP (Gemini Master Process) is a component provided by Gemini that provides most of the Gemini integration features and functionalities. It basically transforms messages sent through the GIAPI-GLUE to the Telescope system components. The language-specific glue is the API layer used by the builder to implement the integration functionality.  For more details of the GMP and GIAPI-GLUE it is recommended to read the documents "GIAPI Design and Use" and "GIAPI C++ Language Glue API". The below image depicts a Simple Instrument architecture to communicate with the Telescope.
```
Comment (cquiroz) [inline README.md:4]: `image below`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+<p align="center">
+<img  align="center" src="https://raw.githubusercontent.com/gemini-hlsw/gmp/igrins2/images/SimpleGMPArhictecture.jpg">
+</p>
+
+# Desployment from binaries. 
+It is possible deploying the GMP and GIAPI-GLUEc++ from their binaries. Gemini recommend uses Rocky 8 or Centos 7 because these are the Operating Systems supported by GEMINI and for which the GIAPI-GLUEc++ binaries have been compiled.
```
Comment (cquiroz) [inline README.md:11]: We may want to make it `giapi-glue C++` everywhere?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+<p align="center">
+<img  align="center" src="https://raw.githubusercontent.com/gemini-hlsw/gmp/igrins2/images/SimpleGMPArhictecture.jpg">
+</p>
+
+# Desployment from binaries. 
+It is possible deploying the GMP and GIAPI-GLUEc++ from their binaries. Gemini recommend uses Rocky 8 or Centos 7 because these are the Operating Systems supported by GEMINI and for which the GIAPI-GLUEc++ binaries have been compiled. 
+
+If you want to deploy the system in other platform, you should compile the external libraries of the [GIAPI-GLUEc++](https://github.com/gemini-hlsw/giapi-glue-cc/tree/develop/external) for you platform. In the previous link is decribed the instrucction executed by GEMINI to CENTOS7 and ROCKY8.
```
Comment (cquiroz) [inline README.md:13]: There are some types and the last paragraph may need a review

File: README.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+## Dependencies 
+The only dependencies we need for GMP and GIAPI-glue are as follows.
+* Java JDK 1.8
+* GCC greater than 4.8.1.
+
+If you are on Rocky 8, you could use the following commands:
+* dnf -y install java-1.8.0-openjdk-devel
```
Comment (cquiroz) [inline README.md:21]: What is `dnf`?

File: README.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+* dnf -y group install "Development Tools"
+
+## Deployment instructions. 
+This section describes the instructions for deploying the GMP and [any-command](https://github.com/gemini-hlsw/giapi-glue-cc/tree/develop/src/examples/any-command.cpp) example implemented in GIAPI-GLUE. The any-command is an example created by GEMINI where developers can see how to implement a command subscriber using the GIAPI-GLUEcc library. Then, we are going to use the giapi-tester to send a command to the any-command example, emulating how any client of the system could send a command to a subscriber. There are more examples in the [src/examples](https://github.com/gemini-hlsw/giapi-glue-cc/tree/develop/src/examples) folder of the GIAPI-GLUEcc repository. 
+
+For this example, the GMP and the any-command example will be deployed on the same machine. If you need deploy them on different machines, you have to modify the gmp.properties which are located in the src/examples folder.
```
Comment (cquiroz) [inline README.md:27]: Mention that you need to set the address where the gmp is running

File: README.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+> cd src/examples
+> If you deployed the GMP on other server or machine, you have to edit the gmp.properties file and specifying the IP or hostname of the server where the GMP was deployed (modify the gmp.hostname field)
+> ./any-command
+```
+
+### GIAPI-TESTER
+Open a terminal or console to execute the following commands. 
+```
+> cd gmp-server-0.2.0/bin
+> java -jar giapi-tester.jar -sc OBSERVE -activity START -config DATA_LABEL=S11172022S001.fits
```
Comment (cquiroz) [inline README.md:53]: I suggest we use something else than OBSERVE which is often different, perhaps an apply?

**Review by cquiroz — COMMENTED** [review]
Thanks a lot, some comments inline

File: README.md (hunk 6 above)
Comment (framos-gemini) [reply README.md:21]: https://en.wikipedia.org/wiki/DNF_(software)

File: README.md (hunk 1 above)
Comment (framos-gemini) [reply README.md:18]: GCC 4.8.1 was the first feature-complete implementation of the 2011 C++ standard, previously known as C++0x. Therefore, the GIAPI-glue supports any GCC compiler greater than the 4.8.1 version.

File: README.md (hunk 4 above)
Comment (framos-gemini) [reply README.md:11]: It would be possible but we will have to compile for the specific platform.

File: README.md (hunk 8 above)
Comment (framos-gemini) [reply README.md:53]: Ok. I added the apply command too.

**Pull request closed without merging**